scrapy/http/response/json.py

```diff
@@ -0,0 +1,23 @@
+"""
+This module implements the JsonResponse class which adds encoding
+discovering through JSON encoding declarations to the TextResponse class.
+
+See documentation in docs/topics/request-response.rst
+"""
+
+import json
+
+from scrapy.http.response.text import TextResponse
+
+
+class JsonResponse(TextResponse):
+
+    def __init__(self, *args, **kwargs):
+        self._json = None
+        super(JsonResponse, self).__init__(*args, **kwargs)
+
+    def json(self):
+        """Returns the JSON-encoded body deserialized into a Python object"""
+        if self._json is None:
+            self._json = json.loads(self.text)
```

Comment:
:lipstick:

Because `null` is a valid JSON document that would be translated into Python as `None`, such documents would always be reloaded.

It is not that much of a issue, but it may be better to use a custom `object()` instance (e.g. define `_NONE = object()` or similar outside the class, and use `_NONE` instead of `None` as default value for `_json` and for the `if` statement). This approach is a common pattern in Python and it has a name, but I cannot remember it at the moment :man_facepalming: